### PR TITLE
Add surface type for image processing.

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michal Zientkiewicz <michalz@nvidia.com>

### CODE REVIEW

File: dali/kernels/imgproc/surface.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  int width, height, channels, pixel_stride, row_stride, channel_stride;
+  __host__ __device__ constexpr T &operator()(int x, int y, int c = 0) const {
+    return data[y * row_stride + x * pixel_stride + c * channel_stride];
+  }
+};
+
+template <typename T, typename Storage>
+constexpr Surface2D<T> as_surface_HWC(const TensorView<Storage, T, 3> &t) {
+  return { t.data,
+    static_cast<int>(t.shape[1]), static_cast<int>(t.shape[0]), static_cast<int>(t.shape[2]),
```
Comment (jantonguirao) [inline dali/kernels/imgproc/surface.h]: would prefer to see which member  you are initializing, instead of going up to the definition to find it

Comment (Kh4L) [reply dali/kernels/imgproc/surface.h]: I agree
```suggestion
    .pixel_stride = static_cast<int>(t.shape[1]), .row_stride = static_cast<int>(t.shape[0]), .channel_stride = static_cast<int>(t.shape[2]),
```
would be more readable here

**Comment by mzient** [comment]
Build #630028 - passed (the relevant parts; others still running).

File: dali/kernels/imgproc/surface.h (hunk 1 above)
Comment (mzient) [reply dali/kernels/imgproc/surface.h]: Unfortunately, deisgnated initializers are part of C, not C++. They are added in C++17 or even 20.

Comment (mzient) [reply dali/kernels/imgproc/surface.h]: ...and adding constructors will make the type non-POD, which I want to avoid.

Comment (jantonguirao) [reply dali/kernels/imgproc/surface.h]: can we do
```
Surface2D<...> surface;
surface.pixel_stride = ...
surface.row_stride = ...
...
```
or at least break it into lines and put a comment at the end of the line of what is what?

Comment (Kh4L) [reply dali/kernels/imgproc/surface.h]: non-trivial designated initializers are not supported prior to C++17, but trivial designated initializers seems to be supported (and implemented in both g++ and clang) from 11

Comment (mzient) [reply dali/kernels/imgproc/surface.h]: We can't do what you propose in C++11 (no longer constexpr). I can break lines.

Comment (Kh4L) [reply dali/kernels/imgproc/surface.h]: @mzient non-trivial designated initializers are not supported prior to C++17, but trivial designated initializers seems to be supported (and implemented in both g++ and clang) from 11

**Pull request merged**